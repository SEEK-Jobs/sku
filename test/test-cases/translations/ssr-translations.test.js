/* eslint-disable jest/expect-expect */
const path = require('path');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const startAssetServer = require('../../utils/assetServer');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const waitForUrls = require('../../utils/waitForUrls');
const appDir = path.resolve(__dirname, 'app');
const skuStartConfig = require('./app/sku.config');

const getTestConfig = (skuConfig) => ({
  backendUrl: `http://localhost:${skuConfig.serverPort}`,
  targetDirectory: path.join(__dirname, skuConfig.target),
});

describe('ssr translations', () => {
  let server;
  const { backendUrl } = getTestConfig(skuStartConfig);

  beforeAll(async () => {
    server = await runSkuScriptInDir('start-ssr', appDir);
    await waitForUrls(backendUrl);
  });

  afterAll(async () => {
    await server.kill();
  });
  it('should render en', async () => {
    const app = await getAppSnapshot(`${backendUrl}/en`);
    expect(app).toMatchSnapshot();
  });

  it('should render fr', async () => {
    const app = await getAppSnapshot(`${backendUrl}/fr`);
    expect(app).toMatchSnapshot();
  });
});
