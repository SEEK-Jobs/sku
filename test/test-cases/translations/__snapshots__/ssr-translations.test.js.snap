// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr translations should render en 1`] = `
SCRIPTS: Array [
  "http://localhost:8313/runtime.js",
  "http://localhost:8313/vendors~main.js",
  "http://localhost:8313/main.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`ssr translations should render fr 1`] = `
SCRIPTS: Array [
  "http://localhost:8313/runtime.js",
  "http://localhost:8313/vendors~main.js",
  "http://localhost:8313/main.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Bonjour sku
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
